scripts/gen_vimdoc.py

```diff
@@ -169,6 +175,12 @@
             'runtime/lua/vim/secure.lua',
             'runtime/lua/vim/version.lua',
             'runtime/lua/vim/_inspector.lua',
+            'runtime/lua/vim/_meta/builtin.lua',
+            'runtime/lua/vim/_meta/diff.lua',
+            'runtime/lua/vim/_meta/mpack.lua',
+            'runtime/lua/vim/_meta/json.lua',
+            'runtime/lua/vim/_meta/misc.lua',
+            'runtime/lua/vim/_meta/spell.lua',
```

Comment:
If the impl is not in Lua we don't really need the separate modules. If we plan to add actual code to these modules then they shouldn't be in `_meta/`, right? 

- I would just put most of these in normal (non-"_meta") modules.
- Instead of `builtin.lua` and `misc.lua`, the oddballs could live in `_meta.lua` or `_editor.lua`? Because `builtin.lua` and `misc.lua` don't sound like meaningful names.

```suggestion
            'runtime/lua/vim/diff.lua',
            'runtime/lua/vim/mpack.lua',
            'runtime/lua/vim/json.lua',
            'runtime/lua/vim/spell.lua',
```

Reply:
These modules are marked with the `@meta` annotation so they shouldn't ever contain any impls. The main motivation is to provide type information for these functions instead of relying on neodev.

I've only put these into separate files as that's the only way I could get the sections in lua.txt, as they were before. Is there another way to achieve this?

Reply:
> I've only put these into separate files as that's the only way I could get the sections in lua.txt, as they were before. Is there another way to achieve this?

Not really.

>> I would just put most of these in normal (non-"_meta") modules.
> 
> Why? ...  Including them in files that get loaded does come with a cost.

I was thinking we might add Lua impl stuff in some of these files potentially. But, didn't know these meta files are ignored by the loader somewhere.

Not a blocker anyways, this PR is a nice gain.